src/lib/components/references/bibleTextReferenceFetcher.ts

```diff
@@ -20,191 +20,108 @@ export const fetchAndFormat = async (
     language: Language,
     passedBibleId?: number
 ): Promise<BibleTextsReference | null> => {
+    if (startVerse > endVerse) {
+        log.exception(new Error(`Unexpected verseIds: startVerse: ${startVerse}; endVerse: ${endVerse}`));
+        return null;
+    }
+
     // ensure we always include verse 0 along with verse 1 if verse 0 exists
     if (startVerse % 1000 === 1) {
         startVerse -= 1;
     }
 
+    // if no Bible was passed then get the default Bible for the language with a fallback of 1
     const bibleId =
         !passedBibleId || passedBibleId === 0
             ? ((await fetchLanguageDefaultBible(language.id))?.id ?? 1)
             : passedBibleId;
 
-    const [bookTexts, versificationMappings] = await Promise.all([
-        fetchBiblePassages(startVerse, endVerse, bibleId),
-        fetchBibleVersification(startVerse, endVerse, bibleId),
-    ]);
-
-    if (!bookTexts || bookTexts.length === 0) {
-        return null;
-    }
-
-    // --- return values ---
-    let verseDisplayName = '';
-    let passageBookTexts: BibleBookTexts[] = [];
-
+    const versificationMappings = await fetchBibleVersification(startVerse, endVerse, bibleId);
     const passageHasDifferentBaseMappings = doesPassageHaveDifferentBaseMappings(
         startVerse,
         endVerse,
         versificationMappings
     );
 
-    // --- prepare start and end passage verses ---
-    const hasMultipleVerses = startVerse !== endVerse;
-    const parsedStartVerse = parseVerseId(startVerse);
-    const parsedEndVerse = hasMultipleVerses ? parseVerseId(endVerse) : null;
+    let targetBibleStartVerseId = startVerse;
+    let targetBibleEndVerseId = endVerse;
 
     // This logic assumes that all returned mappings are an exact range in the target Bible.  This is not necessarily true but is easier to reason about.
     // In the future we may need to update this logic to handle more complex mappings if real-world scenarios arise where mappings are non-consecutive verses.
     if (passageHasDifferentBaseMappings) {
-        if (!hasMultipleVerses) {
```

Comment:
I collapsed all four of the code branches (different mappings with multiple verses, difference mappings with a single verse, same mappings with multiple verses, same mappings with a single verse) all into a single code flow.  This makes error handling easier because I didn't have to copy/paste it into multiple branches.